### PR TITLE
Add getNrgPrice Java API call

### PR DESCRIPTION
**Pull request by aionick**
## Description

Please include a brief summary of the change that this pull request proposes. Include any relevant motivation and context. List any dependencies required for this change.

- The NrgOracle class provides a method to recommend a user an energy price to use, so far only web3 API gives access to this and not Java API. I've added a call to the Java API to access this value. See my <a href="https://github.com/aionnetwork/aion_api/pull/24">other PR</a> in aion_api project that implements the rest of this functionality.

Fixes Issue # N/A.

## Type of change

Insert **x** into the following checkboxes to confirm (eg. [x]):
- [ ] Bug fix.
- [x] New feature.
- [ ] Breaking change (a fix or feature that causes existing functionality to not work as expected).
- [ ] Requires documentation update.

## Testing

Please describe the tests you used to validate this pull request. Provide any relevant details for test configurations as well as any instructions to reproduce these results.

- I added a test to the BaseAPITest in the aion_api project for this. Ran through it in debugger a few times.

## Verification

Insert **x** into the following checkboxes to confirm (eg. [x]):
- [x] I have self-reviewed my own code and conformed to the style guidelines of this project.
- [x] New and existing tests pass locally with my changes.
- [x] I have added tests for my fix or feature.
- [x] I have made appropriate changes to the corresponding documentation.
- [x] My code generates no new warnings.
- [x] Any dependent changes have been made.

### CODE REVIEW

**Comment by qoire** [comment]
@aionick formatter seems to be taking up most of the code changes, possible for you to point out the specific sections with meaning edits?

Also some of the formatting is problematic especially when it concerns lambdas

**Comment by aionick** [comment]
My bad I had written that out and meant to comment but forgot. ApiAion0 is the only thing I manually changed here, lines 633-661, the case: case Message.Funcs.f_getNrgPrice_VALUE

**Comment by aionick** [comment]
@qoire The lambda formatting is really annoying, looks like there was an issue raised about this on google's githib <a href="https://github.com/google/google-java-format/issues/19">here</a> but no response for a few months. I'm tweaking the script to ignore our license atm and I'll see if I can find a workaround for the lambdas

File: modApiServer/src/org/aion/api/server/pb/ApiAion0.java [hunk 1]
```diff
@@ -307,1514 +364,2092 @@ public ApiAion0(IAionChain ac) {
...
-            try {
-                Message.req_getBalance req = Message.req_getBalance.parseFrom(data);
+                    // instantiate nrg price oracle
+                    IAionBlockchain bc = (IAionBlockchain) AionImpl.inst().getBlockchain();
+                    long nrgPriceDefault = CfgAion.inst().getApi().getNrg().getNrgPriceDefault();
+                    long nrgPriceMax = CfgAion.inst().getApi().getNrg().getNrgPriceMax();
+                    NrgOracle.Strategy oracleStrategy = NrgOracle.Strategy.SIMPLE;
+
+                    this.nrgOracle = new NrgOracle(bc, nrgPriceDefault, nrgPriceMax, oracleStrategy);
+                    long nrg = this.getRecommendedNrgPrice();
```
Comment (AionJayT) [inline modApiServer/src/org/aion/api/server/pb/ApiAion0.java]: Web3Aion has the same logic, please considering move the NrgOracle logic in both web3 and java api to the abstract class ApiAion

**Pull request merged**